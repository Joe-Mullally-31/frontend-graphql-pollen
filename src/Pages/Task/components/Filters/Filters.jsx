import React from "react";
import { makeStyles } from "@material-ui/core/styles";
import Accordion from "@material-ui/core/Accordion";
import AccordionSummary from "@material-ui/core/AccordionSummary";
import AccordionDetails from "@material-ui/core/AccordionDetails";
import Typography from "@material-ui/core/Typography";
import ExpandMoreIcon from "@material-ui/icons/ExpandMore";
import { Slider } from "@material-ui/core";
import { connect } from "react-redux";
import {
  updateCapRange,
  updatePointsRange,
  updateQuantityRange,
} from "../../actions";

const useStyles = makeStyles((theme) => ({
  root: {
    padding: "16px 16px",
  },
  heading: {
    fontSize: theme.typography.pxToRem(15),
    fontWeight: theme.typography.fontWeightRegular,
  },
  accordionDetails: {
    width: "100%",
    padding: "15px 20px",
  },
}));

export const Filters = ({ dispatch, quantityRange, pointsRange, capRange }) => {
  const classes = useStyles();

  const handleQuantityChange = (event, newQuantity) => {
    dispatch(updateQuantityRange(newQuantity[0], newQuantity[1]));
  };

  const handlePointsChange = (event, newPoints) => {
    dispatch(updatePointsRange(newPoints[0], newPoints[1]));
  };

  const handleCapChange = (event, newCaps) => {
    dispatch(updateCapRange(newCaps[0], newCaps[1]));
  };

  const quantityMarks = [
    { value: 0, label: "0" },
    { value: 100, label: "100" },
  ];
  const pointsMarks = [
    { value: 0, label: "0" },
    { value: 10000, label: "10,000" },
  ];
  const capMarks = [
    { value: 0, label: "0" },
    { value: 5, label: "5" },
  ];

  return (
    <div className={classes.root}>
      <Accordion>
        <AccordionSummary
          expandIcon={<ExpandMoreIcon />}
          aria-controls="panel1a-content"
          id="panel1a-header"
        >
          <Typography className={classes.heading}>Filters</Typography>
        </AccordionSummary>
        <AccordionDetails>
          <div className={classes.accordionDetails}>
            <Typography id="quantity-slider" gutterBottom>
              Quantity
            </Typography>
            <Slider
              min={0}
              max={100}
              value={quantityRange}
              onChange={handleQuantityChange}
              valueLabelDisplay="auto"
              aria-labelledby="quantity-slider"
              marks={quantityMarks}
              className="quantitySlider"
            />
            <Typography id="points-slider" gutterBottom>
              Points
            </Typography>
            <Slider
              min={0}
              max={10000}
              value={pointsRange}
              onChange={handlePointsChange}
              valueLabelDisplay="auto"
              aria-labelledby="points-slider"
              marks={pointsMarks}
              step={500}
              className="pointsSlider"
            />
            <Typography id="cap-slider" gutterBottom>
              Cap
            </Typography>
            <Slider
              min={0}
              max={5}
              value={capRange}
              onChange={handleCapChange}
              valueLabelDisplay="auto"
              aria-labelledby="cap-slider"
              marks={capMarks}
              step={1}
              className="capSlider"
            />
          </div>
        </AccordionDetails>
      </Accordion>
    </div>
  );
};

const mapStateToProps = (state) =>
  Object({
    quantityRange: [state.quantityRange.min, state.quantityRange.max],
    pointsRange: [state.pointsRange.min, state.pointsRange.max],
    capRange: [state.capRange.min, state.capRange.max],
  });

export default connect(mapStateToProps)(Filters);
