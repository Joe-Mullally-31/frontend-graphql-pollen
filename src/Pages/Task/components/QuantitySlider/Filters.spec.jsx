import React from "react";
import { shallow } from "enzyme";
import Filters from "./Filters.jsx";
import * as fireEvent from "react-dom/test-utils";

describe("<Filters />", () => {
  let component;
  let instance;
  const props = {};
  beforeEach(() => {
    component = shallow(<Filters {...props} />);
  });

  it("renders with props", () => {
    expect(component).toMatchSnapshot();
    instance = component.instance();
  });

  it("updates quantity when quantity slider changes", () => {
    expect(component.find(".quantitySlider").props().value).toEqual([0, 100]);

    component.find(".quantitySlider").props().onChange(undefined, [25, 80]);

    expect(component.find(".quantitySlider").props().value).toEqual([25, 80]);
  });

  it("updates points when points slider changes", () => {
    expect(component.find(".pointsSlider").props().value).toEqual([0, 10000]);

    component.find(".pointsSlider").props().onChange(undefined, [25, 80]);

    expect(component.find(".pointsSlider").props().value).toEqual([25, 80]);
  });

  it("updates cap when cap slider changes", () => {
    expect(component.find(".capSlider").props().value).toEqual([0, 5]);

    component.find(".capSlider").props().onChange(undefined, [25, 80]);

    expect(component.find(".capSlider").props().value).toEqual([25, 80]);
  });
});
