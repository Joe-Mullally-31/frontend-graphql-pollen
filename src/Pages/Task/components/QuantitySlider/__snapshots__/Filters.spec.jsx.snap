// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Filters /> renders with props 1`] = `
<div
  className="makeStyles-root-1"
>
  <WithStyles(ForwardRef(Accordion))>
    <WithStyles(ForwardRef(AccordionSummary))
      aria-controls="panel1a-content"
      expandIcon={<Memo />}
      id="panel1a-header"
    >
      <WithStyles(ForwardRef(Typography))
        className="makeStyles-heading-2"
      >
        Filters
      </WithStyles(ForwardRef(Typography))>
    </WithStyles(ForwardRef(AccordionSummary))>
    <WithStyles(ForwardRef(AccordionDetails))>
      <div
        className="makeStyles-accordionDetails-3"
      >
        <WithStyles(ForwardRef(Typography))
          gutterBottom={true}
          id="quantity-slider"
        >
          Quantity
        </WithStyles(ForwardRef(Typography))>
        <WithStyles(ForwardRef(Slider))
          aria-labelledby="quantity-slider"
          className="quantitySlider"
          marks={
            Array [
              Object {
                "label": "0",
                "value": 0,
              },
              Object {
                "label": "100",
                "value": 100,
              },
            ]
          }
          max={100}
          min={0}
          onChange={[Function]}
          value={
            Array [
              0,
              100,
            ]
          }
          valueLabelDisplay="auto"
        />
        <WithStyles(ForwardRef(Typography))
          gutterBottom={true}
          id="points-slider"
        >
          Points
        </WithStyles(ForwardRef(Typography))>
        <WithStyles(ForwardRef(Slider))
          aria-labelledby="points-slider"
          className="pointsSlider"
          marks={
            Array [
              Object {
                "label": "0",
                "value": 0,
              },
              Object {
                "label": "10,000",
                "value": 10000,
              },
            ]
          }
          max={10000}
          min={0}
          onChange={[Function]}
          step={500}
          value={
            Array [
              0,
              10000,
            ]
          }
          valueLabelDisplay="auto"
        />
        <WithStyles(ForwardRef(Typography))
          gutterBottom={true}
          id="cap-slider"
        >
          Cap
        </WithStyles(ForwardRef(Typography))>
        <WithStyles(ForwardRef(Slider))
          aria-labelledby="cap-slider"
          className="capSlider"
          marks={
            Array [
              Object {
                "label": "0",
                "value": 0,
              },
              Object {
                "label": "5",
                "value": 5,
              },
            ]
          }
          max={5}
          min={0}
          onChange={[Function]}
          step={1}
          value={
            Array [
              0,
              5,
            ]
          }
          valueLabelDisplay="auto"
        />
      </div>
    </WithStyles(ForwardRef(AccordionDetails))>
  </WithStyles(ForwardRef(Accordion))>
</div>
`;
